File: 송하연/9월 5주차/최고의집합.java
import java.util.*;

class Solution {
    public int[] solution(int n, int s) {
        List<Integer> result = new ArrayList<>();
        
        int num = s/n;
        int rest = s%n;
        
        if(n>s) {
            return new int[]{-1};
        }
       
        int[] answer = new int[n];
        Arrays.fill(answer,num);

        for(int i=n-1; i>=0; i--){
            if(rest==0) break;
            answer[i]++;
            rest--;
        }

        return answer;
    }
}

File: 이정선/9월4주차/최고의집합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int s) {
    vector<int> answer;
    
    vector<int> bestList(n);
    
    if(s < n) return {-1};
    
    // s 를 배열에 분할
    for(int i = 0; i < n; i ++) {
        bestList[i] += s/n;
    }
    
    for(int i = 0; i < s%n; i ++) {
        bestList[n-1-i] += 1;
    }
    
    answer = bestList;
    return answer;
}
